bgpd/bgp_route.c

```diff
@@ -614,10 +615,13 @@ static int bgp_path_info_cmp(struct bgp *bgp, struct bgp_path_info *new,
 	newattr = new->attr;
 	existattr = exist->attr;
 
+	new_p = bgp_dest_get_prefix(new->net);
+
 	/* For EVPN routes, we cannot just go by local vs remote, we have to
 	 * look at the MAC mobility sequence number, if present.
 	 */
-	if (safi == SAFI_EVPN) {
+	if ((safi == SAFI_EVPN)
+	    && (new_p->u.prefix_evpn.route_type != BGP_EVPN_IP_PREFIX_ROUTE)) {
```

Comment:
Below code applies only to Type-2, and shouldn't be applied for Type-3, Type-4, Type-5 and Type-1. So I think better fix would be to check if it is Type-2 route or not.

Some thing like below:
if (safi == SAFI_EVPN &&
	   new_p->u.prefix_evpn.route_type == BGP_EVPN_MAC_IP_ROUTE) {

Reply:
Thanks for this feedback - pushed this update.